docs/content/envoy-tutorial-istio.md

```diff
@@ -0,0 +1,196 @@
+    }
+    
+    required_roles[r] {
+        perm := role_perms[r][_]
+        perm.method = http_request.method
+        perm.path = http_request.path
+    }
+    
+    user_name = parsed {
+        [_, encoded] := split(http_request.headers.authorization, " ")
+        [parsed, _] := split(base64url.decode(encoded), ":")
+    }
+    
+    user_roles = {
+        "alice": ["guest"],
+        "bob": ["admin"]
+    }
+    
+    role_perms = {
+        "guest": [
+            {"method": "GET",  "path": "/productpage"},
+        ],
+        "admin": [
+            {"method": "GET",  "path": "/productpage"},
+            {"method": "GET",  "path": "/api/v1/products"},
+        ],
+    }
+    ```
+
+    OPA is configured to query for the `data.istio.authz.allow`
+    decision. If the response is `true` the operation is allowed, otherwise the
+    operation is denied. Sample input received by OPA is shown below:
+
+    ```live:example:query:hidden
+    data.istio.authz.allow
+    ```
+  
+    ```live:example:input
+    {
+        "attributes": {
+            "request": {
+                "http": {
+                    "method": "GET",
+                    "path": "/productpage",
+                    "headers": {
+                        "authorization": "Bearer YWxpY2U6cGFzc3dvcmQ="
+                    }
+                }
+            }
+        }
+    }
+    ```
+
+    With the input value above, the answer is:
+    
+    ```live:example:output
+    ```
+
+    An example of the complete input received by OPA can be seen [here](https://github.com/open-policy-agent/opa-envoy-plugin/tree/master/examples/istio#example-input).
+
+    > In typical deployments the policy would either be built into the OPA container
+    > image or it would fetched dynamically via the [Bundle
+    > API](https://www.openpolicyagent.org/docs/latest/bundles/). ConfigMaps are
+    > used in this tutorial for test purposes.
+  
+### 2. Enable automatic injection of the Istio Proxy and OPA-Envoy sidecars in the namespace where the app will be deployed, e.g., `default`
+
+```bash
+kubectl label namespace default opa-istio-injection="enabled"
+kubectl label namespace default istio-injection="enabled"
+```
+
+### 3. Deploy the BookInfo application and make it accessible outside the cluster
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/istio/istio/master/samples/bookinfo/platform/kube/bookinfo.yaml
+```
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/istio/istio/master/samples/bookinfo/networking/bookinfo-gateway.yaml
+```
+
+### 4. Set the `GATEWAY_URL` environment variable in your shell to the public IP/port of the Istio Ingress gateway
+
+**minikube**:
+
+```bash
+export INGRESS_PORT=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="http2")].nodePort}')
+export INGRESS_HOST=$(minikube ip)
+export GATEWAY_URL=$INGRESS_HOST:$INGRESS_PORT
+echo $GATEWAY_URL
+```
+
+**minikube (example)**:
+
+```bash
```

Comment:
[nit] We've got the same thing on the standalone tutorial, but the `bash` here makes the markup look funny:
![image](https://user-images.githubusercontent.com/870638/108361302-93c33400-71f2-11eb-8314-f6fc91a7d6f3.png)

Let's go with plain three backticks, I suppose?

Reply:
Updated in both tutorials.